core/storage/learner_group/gae_models.py

```diff
@@ -0,0 +1,291 @@
+                    'subtopics_page_ids':
+                        learner_group_model.subtopics_page_ids,
+                    'story_ids': learner_group_model.story_ids
+                }
+        return user_data
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether LearnerGroupModel contains data of a given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return (
+            cls.query(datastore_services.any_of(
+                cls.students_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitators_user_ids == user_id
+            )).get(keys_only=True) is not None
+        )
+
+    @classmethod
+    def apply_deletion_policy(cls, user_id: str) -> None:
+        """Delete all LearnerGroupModel instances associated with the
+        user.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to delete.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.students_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitators_user_ids == user_id
+        ))
+
+        for learner_group_model in found_models:
+            # If the user is a student, delete the user from the
+            # students_user_ids list.
+            if user_id in learner_group_model.students_user_ids:
+                learner_group_model.students_user_ids.remove(user_id)
+                learner_group_model.update_timestamps()
+                learner_group_model.put()
+
+            # If the user has been invited to join the group, delete the
+            # user from the invited_user_ids list.
+            elif user_id in learner_group_model.invited_user_ids:
+                learner_group_model.invited_user_ids.remove(user_id)
+                learner_group_model.update_timestamps()
+                learner_group_model.put()
+
+            # If the user is the facilitator of the group and there are
+            # more then one facilitators_user_ids, delete the user from the
+            # facilitators_user_ids list.
+            elif user_id in learner_group_model.facilitators_user_ids and (
+                    len(learner_group_model.facilitators_user_ids) > 1):
+                learner_group_model.facilitators_user_ids.remove(user_id)
+                learner_group_model.update_timestamps()
+                learner_group_model.put()
+
+            # If the user is the facilitator of the group and there is
+            # only one facilitator_user_id, delete the group.
+            elif user_id in learner_group_model.facilitators_user_ids and (
+                    len(learner_group_model.facilitators_user_ids) == 1):
+                learner_group_model.delete()
```

Comment:
Would deleting the model affect other LearnerGroupUser models, since they would have fields pointing to invalid group IDs? You might need a separate function for this.

Reply:
Yeah, added that.